tools/stdlib_diff.py

```diff
@@ -0,0 +1,164 @@
+"""Simple command line utility to help finding the differences between
+ConfigUpdater and ConfigParser.
+
+The output of this script is similar to the output produced by
+`git format-patch`. This means that programs like `delta` or `bat`
+can be used to add some format highlighting to the output. Examples:
+
+    %(prog)s | delta --side-by-side
+    %(prog)s | bat -l diff
+
+Alternatively https://github.com/megatops/PatchViewer can also be used.
+
+[delta]: https://github.com/dandavison/delta
+[bat]: https://github.com/sharkdp/bat
+[PathViewer]:  https://github.com/megatops/PatchViewer
+"""
+
+import argparse
+import os.path
+import sys
+from configparser import ConfigParser, SectionProxy
+from dataclasses import dataclass
+from difflib import SequenceMatcher
+from inspect import getmembers, getsourcefile, getsourcelines
+from typing import Iterator, Sequence, Type
+
+try:
+    import configupdater  # noqa
+except ImportError:
+    repo = os.path.dirname(os.path.dirname(__file__))
+    sys.path.append(os.path.join(repo, "src"))
+
+from configupdater import ConfigUpdater, Parser, Section
+from configupdater.document import Document
+
+COMPARISONS = [
+    (Parser, ConfigParser),
+    (ConfigUpdater, ConfigParser),
+    (Document, ConfigParser),
+    (Section, SectionProxy),
+]
+
+
+def diff_all(numlines: int) -> str:
+    return "".join(
+        patch
+        for (target, orig) in COMPARISONS
+        for patch in diff_class(orig, target, numlines)
+    )
+
+
+def diff_class(orig_cls: Type, changed_cls: Type, numlines: int) -> Iterator[str]:
+    diff_fragments = (
+        diff_member(name, orig_cls, changed_cls, numlines)
+        for name, _ in getmembers(changed_cls)
+        if name != "__init__" and name in changed_cls.__dict__
+    )
+    for patch in diff_fragments:
+        yield from patch
+
+
+def diff_member(
+    name: str, orig_cls: Type, changed_cls: Type, numlines: int
+) -> Iterator[str]:
+    orig = getattr(orig_cls, name, None)
+    changed = getattr(changed_cls, name, None)
+
+    if orig is None or changed is None:
+        return None
+
+    if not (callable(orig) and callable(changed)):
+        # Not a method
+        return None
+
+    try:
+        orig_code = CodeInfo.inspect(orig)
+        changed_code = CodeInfo.inspect(changed)
+    except TypeError:
+        # buitins will fail if inspected
+        return None
+
+    if orig_code.file.endswith("collections_abc.py"):
```

Comment:
Here I decided to ignore when the original class uses `collections.abc.Mapping,MutableMapping,...`

But maybe I should not?